controllers/inventory_controller.go

```diff
@@ -0,0 +1,73 @@
+package controllers
+
+import (
+	"context"
+
+	"github.com/shipwright-io/build/pkg/apis/build/v1alpha1"
+	"github.com/shipwright-io/triggers/pkg/inventory"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+// InventoryReconciler reconciles Build instances on the Inventory.
+type InventoryReconciler struct {
+	client.Client                 // kubernetes client
+	Scheme        *runtime.Scheme // shared scheme
+	Clock                         // local clock instance
+
+	buildInventory *inventory.Inventory // local build triggers database
+}
+
+//+kubebuilder:rbac:groups=shipwright.io,resources=builds,verbs=get;list;watch
+
+// Reconcile reconciles Build instances reflecting it's status on the Inventory.
+func (r *InventoryReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	logger := log.FromContext(ctx)
+
+	var b v1alpha1.Build
+	if err := r.Get(ctx, req.NamespacedName, &b); err != nil {
+		if !errors.IsNotFound(err) {
+			logger.Error(err, "Unable to fetch Build, removing from the Inventory")
+		}
+		r.buildInventory.Remove(req.NamespacedName)
+		return ctrl.Result{}, client.IgnoreNotFound(err)
```

Comment:
I suggest to not remove the build when the error is not IsNotFound.

```suggestion
		if !errors.IsNotFound(err) {
			logger.Error(err, "Unable to fetch Build, removing from the Inventory")
			return ctrl.Result{}, client.IgnoreNotFound(err)
		} else {
			r.buildInventory.Remove(req.NamespacedName)
		}
```

Reply:
In this case, @SaschaSchwarze0, we need to reflect the current state of the informer on the Inventory, when the informer can't find the instance the Inventory must have the same state.

Please consider [this example](https://book.kubebuilder.io/cronjob-tutorial/controller-implementation.html#1-load-the-cronjob-by-name), the object may change in the next interaction, so the controller can just wait for the next event to reconcile the final object state.